lib/app/features/feed/create_post/views/pages/post_form_modal/hooks/use_nsfw_validation.dart

```diff
@@ -0,0 +1,61 @@
+// SPDX-License-Identifier: ice License 1.0
+
+import 'package:flutter/material.dart';
+import 'package:flutter_hooks/flutter_hooks.dart';
+import 'package:hooks_riverpod/hooks_riverpod.dart';
+import 'package:ion/app/features/nsfw/providers/media_nsfw_checker_notifier.m.dart';
+import 'package:ion/app/services/media_service/media_service.m.dart';
+
+void useNsfwValidation({
+  required List<MediaFile> mediaFiles,
+  required MediaFile? videoFile,
+  required WidgetRef ref,
+}) {
+  final mediaSignature = mediaFiles.map((f) => f.path).join(',');
+
+  // Handle image files
+  useEffect(
+    () {
+      WidgetsBinding.instance.addPostFrameCallback((_) async {
+        final mediaChecker = ref.read(mediaNsfwCheckerNotifierProvider.notifier);
+        final mediaCheckerState = ref.read(mediaNsfwCheckerNotifierProvider);
+
+        if (mediaFiles.isNotEmpty) {
+          final convertedMediaFiles = await ref
+              .read(mediaServiceProvider)
+              .convertAssetIdsToMediaFiles(ref, mediaFiles: mediaFiles);
+
+          await mediaChecker.checkMediaForNsfw(convertedMediaFiles);
+        } else {
+          if (!mediaCheckerState.isEmpty) {
+            // The case when we removed last media, but we still have that result kept in the state
+            await mediaChecker.checkMediaForNsfw([]);
+          }
+        }
+      });
+      return null;
+    },
+    [mediaSignature],
+  );
+
+// Handle video files
+  useEffect(
+    () {
+      WidgetsBinding.instance.addPostFrameCallback((_) async {
+        final mediaChecker = ref.read(mediaNsfwCheckerNotifierProvider.notifier);
+        final mediaCheckerState = ref.read(mediaNsfwCheckerNotifierProvider);
+
+        if (videoFile != null) {
+          await mediaChecker.checkMediaForNsfw([videoFile]);
+        } else {
+          if (!mediaCheckerState.isEmpty) {
+            // The case when we removed last video, but we still have that result kept in the state
+            await mediaChecker.checkMediaForNsfw([]);
```

Comment:
Can u please describe what is the case when we need to call the function with the empty list

Reply:
Ah, got it. Maybe just add a `mediaChecker.reset` method or something like that? That will be more obvious and won't require a comment to clarify ?